### PR TITLE
Adds support for Table storage

### PR DESCRIPTION
**Pull request by simongdavies**
This change adds support for using Azure Table storage instead of Azure Blob storage.

Full details on usage in the README.

### CODE REVIEW

**Review by vdice — COMMENTED** [review]
Tested this with my storage account and works great!

File: .vscode/tasks.json [hunk 1]
```diff
@@ -22,17 +22,18 @@
       "command": "./.vscode/scripts/runporterinbackground.sh",
       "options": {
         "env": {
-          "PORTER_RUN_PLUGIN_IN_DEBUGGER": "storage.azure.blob",
+          "PORTER_RUN_PLUGIN_IN_DEBUGGER": "storage.azure.table",
           "PORTER_PLUGIN_WORKING_DIRECTORY": "${workspaceRoot}/cmd/azure",
           "PORTER_DEBUGGER_PORT": "2345",
-          "PORTER_HOME": "/home/${env:USER}/.porter"
+          "PORTER_HOME": "/home/${env:USER}/.porter",
+          "AZURE_STORAGE_CONNECTION_STRING":"DefaultEndpointsProtocol=https;AccountName=cnabstategrufuyk7dkalg;AccountKey=H85r3atfaHo08uQz9EcbengVwzK5O0Vc652CUliL6+IlzPMdhw+MIXct2Ec1sjmy3CHzlRBo4A9hMlpgxjAFHg==;EndpointSuffix=core.windows.net"
```
[the remark this answers is not in the document]
Comment (simongdavies) [reply .vscode/tasks.json]: Oh good catch , this is a storage account I used for debugging, its a test account but I rotated the keys and have updated the task.json to remove it.

File: pkg/azure/table/credentials.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+	"github.com/Azure/azure-sdk-for-go/profiles/latest/storage/mgmt/storage"
+	"github.com/Azure/go-autorest/autorest/azure/auth"
+	"github.com/hashicorp/go-hclog"
+	"github.com/pkg/errors"
+)
+
+const ConnectionEnvironmentVariable = "AZURE_STORAGE_CONNECTION_STRING"
+const UserAgent = "porter.azure.storage.plugin"
+
+func GetCredentials(cfg azureconfig.Config, l hclog.Logger) (string, string, error) {
```
Comment (vdice) [inline pkg/azure/table/credentials.go:20]: Perhaps for a follow-up, but would either/both of `GetCredentials` and `GetCredentialsFromCli` be candidates for inclusion into the newly-formed `common` pkg?  There look to be a few variants of these spread across table and blob storage pkgs...

**Review by vdice — APPROVED** [review]

File: pkg/azure/table/credentials.go (hunk 2 above)
Comment (simongdavies) [reply pkg/azure/table/credentials.go:20]: Yeah there are a few things like this that would be good to do, I'll take a look once its merged

**Pull request merged**